lib/src/lib.nr

```diff
@@ -85,12 +83,31 @@ impl<PROOF_LEN, MAX_VALUE_LEN> TrieProof<32, PROOF_LEN, MAX_VALUE_LEN>
             verify_internal_node(&mut extracted_hash, &mut key_ptr, key_nibbles, path, depth, i);
         }
 
-        let mut leaf_node = [0; MAX_TRIE_NODE_LENGTH];
+        self.verify_leaf_storage_node(extracted_hash, key.len(), key_ptr, key_nibbles);
+        
+        true
+    }
+
+    /// Verifies that terminal storage node hashes to extracted hash. Extracts value and verifies it against the argument value.
+    ///
+    /// # Arguments
+    /// * `hash` - Hash extracted from the preceding node
+    /// * `value` - Value to be verified
+    /// * `value_length` - Length of the value
+    /// * `key` - Key to be verified
+    /// * `key_ptr` - Pointer to current nibble in the key
+    /// * `key_nibbles` - Array containing the nibbles of the key
+    /// * `path` - RLP-encoded proof path
+    /// * `depth` - Depth of proof path
+    pub fn verify_leaf_storage_node(self, hash: [u8; KEY_LENGTH], key_len: Field, key_ptr: u64, key_nibbles: [u8; NIBBLE_LENGTH]) 
```

Comment:
Could you rename this to `verify_storage_leaf_node`?

Reply:
Sure, renamed